api/types/transfer/progress.proto

```diff
@@ -0,0 +1,29 @@
+/*
+	Copyright The containerd Authors.
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+*/
+
+syntax = "proto3";
+
+package containerd.v1.types;
+
+option go_package = "github.com/containerd/containerd/api/types/transfer";
+
+message Progress {
+	string event = 1;
+	string name = 2;
+	repeated string parents = 3;
```

Comment:
How can we have multiple parents here? 
Asking this in the context of transferring a blob or a manifest. It will always have a single parent right?

Reply:
A blob could be used by multiple parents. For example the same layer that is used by arm/v6 and arm/v8. The reason this is a pointer up (as opposed to how we normally list children) is because the parents affect where the progress is shown and the parent will always exist as it would be processed first. The reverse isn't always true though since children may be filtered out.